src/directives/mmsDiffAttr.js

```diff
@@ -17,212 +17,166 @@ angular.module('mms.directives')
         var baseDeleted = false;
         var compDeleted = false;
-        if (mmsViewCtrl) {
-            viewOrigin = mmsViewCtrl.getElementOrigin();
-        }
-        scope.options = {
-            editCost: 8
-        };
 
-        /**
-         * @ngdoc function
-         * @name mms.directives.directive:mmsDiffAttr#changeElement
-         * @methodOf mms.directives.directive:mmsDiffAttr
-         *
-         * @description
-         * Change scope for diff when there is a change in commit id
-         */
-        var changeElement = function(newVal, oldVal) {
-            if (!newVal || (newVal == oldVal && ran)) {
-                return;
-            }
-
-            scope.message = '';
-            scope.origElem = '';
-            scope.compElem = '';
-            scope.diffLoading = true;
-            var baseElementId = scope.mmsBaseElementId;
-            var compareElementId = scope.mmsCompareElementId;
-            var baseProjectId = scope.mmsBaseProjectId;
-            var compareProjectId = scope.mmsCompareProjectId;
-            var baseRefId = scope.mmsBaseRefId;
-            var compareRefId = scope.mmsCompareRefId;
-            var baseCommitId = scope.mmsBaseCommitId;
-            var compareCommitId = scope.mmsCompareCommitId;
+        _setupDiff();
 
+        function _setupDiff() {
+            $scope.diffLoading = true;
             baseNotFound = false;
             compNotFound = false;
             baseDeleted = false;
             compDeleted = false;
 
-            if (!compareElementId) {
-                compareElementId = baseElementId;
-            }
-            if (!baseProjectId && viewOrigin) {
-                baseProjectId = viewOrigin.projectId;
-            // } else {
-            //     // return
-            }
-            if (!compareProjectId) {
-                compareProjectId = baseProjectId;
-            }
-            if (!baseRefId && viewOrigin) {
-                baseRefId = viewOrigin.refId;
-            } else if (!baseRefId && !viewOrigin) {
-                baseRefId = 'master';
-            }
-            if (!compareRefId) {
-                compareRefId = baseRefId;
-            }
-            if (!baseCommitId) {
-                baseCommitId = 'latest';
-            }
-            if (!compareCommitId) {
-                compareCommitId = 'latest';
-            }
+            var projectId = $scope.mmsProjectId;
+            var elementId = $scope.mmsElementId;
+
+            var baseRefId = $scope.mmsBaseRefId || 'master';
+            var baseCommitId = $scope.mmsBaseCommitId || 'latest';
+
+            var compareRefId = $scope.mmsCompareRefId || baseRefId;
+            var compareCommitId = $scope.mmsCompareCommitId || 'latest';
+
             if (baseCommitId === compareCommitId) {
-                scope.message = ' Comparing same version.';
-                scope.diffLoading = false;
+                $scope.message = ' Comparing same version.';
```

Comment:
think this was a bug before too, but if both commit ids are latest, need to compare element id, refid, and project id to be sure it's the same

Reply:
make sense. I will add extra check.